sklearn/linear_model/_base.py

```diff
@@ -868,18 +765,12 @@ def _pre_fit(
             # This triggers copies anyway.
             X, y, _ = _rescale_data(X, y, sample_weight=sample_weight)
 
-    # FIXME: 'normalize' to be removed in 1.4
     if hasattr(precompute, "__array__"):
-        if (
-            fit_intercept
-            and not np.allclose(X_offset, np.zeros(n_features))
-            or normalize
-            and not np.allclose(X_scale, np.ones(n_features))
-        ):
+        if fit_intercept and not np.allclose(X_offset, np.zeros(n_features)):
             warnings.warn(
                 (
                     "Gram matrix was provided but X was centered to fit "
-                    "intercept, or X was normalized : recomputing Gram matrix."
+                    "intercept: recomputing Gram matrix."
                 ),
                 UserWarning,
             )
```

Comment:
```suggestion
            )
            # TODO: instead of warning and recomputing, we could just center
            # the user provided Gram matrix a-posteriori (after making a copy
            # when `copy=True`).
```